samza-log4j/src/main/java/org/apache/samza/config/Log4jSystemConfig.java

```diff
@@ -78,7 +81,8 @@ public String getSerdeClass(String name) {
   }
 
   public String getStreamSerdeName(String systemName, String streamName) {
-    String streamSerdeNameConfig = String.format(StreamConfig.MSG_SERDE(), systemName, streamName);
-    return get(streamSerdeNameConfig, null);
+    StreamConfig streamConfig =  new StreamConfig(this);
+    scala.Option<String> option = streamConfig.getStreamMsgSerde(new SystemStream(systemName, streamName));
```

Comment:
I wouldn't say that we don't plan on it. It's just not part of this patch. There are a number of places in the code in which SystemStream is still used (e.g. task.inputs). SAMZA-1118 was filed to go through all the configurations and migrate them to streamId.
When I get back, I'd be happy to start on that one if you think it's MVP.